### PR TITLE
ROC-6821: Make payment optional set in callback

### PR DESCRIPTION
**Pull request by sabahirfan**
### JIRA link (if applicable) ###
[ROC-6821](https://tools.hmcts.net/jira/browse/ROC-6821)

### Change description ###
Make payment and feeAmountInPennies optional as these are being calculated in callback.

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

### CODE REVIEW

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ClaimMapper.java [hunk 1]
```diff
@@ -92,14 +92,15 @@ public void to(Claim claim, CCDCase.CCDCaseBuilder builder) {
...
         claimData.getTimeline().ifPresent(timeline -> timelineMapper.to(timeline, builder));
         claimData.getEvidence().ifPresent(evidence -> evidenceMapper.to(evidence, builder));
+        claimData.getPayment().ifPresent(payment -> paymentMapper.to(payment, builder));
 
-        paymentMapper.to(claimData.getPayment(), builder);
         interestMapper.to(claimData.getInterest(), builder);
         amountMapper.to(claimData.getAmount(), builder);
         claim.getTotalAmountTillDateOfIssue().map(moneyMapper::to).ifPresent(builder::totalAmount);
+        claimData.getFeeAmountInPennies()
+            .ifPresent(feeAmountInPennies -> builder.feeAmountInPennies(feeAmountInPennies.toString()));
```
Comment (andrewwa-kainos) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/ClaimMapper.java]: Preference is to keep each chain item as simple as possible:
```
claimData.getFeeAmountInPennies()
    .map(BigInteger::toString)
    .ifPresent(builder::feeAmountInPennies);
```

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/ClaimData.java [hunk 2]
```diff
@@ -55,8 +54,6 @@
     @NotNull
     private final Amount amount;
 
-    @NotNull
-    @Min(0)
```
Comment (andrewwa-kainos) [inline domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/ClaimData.java]: Don't we still want the `Min(0)`? I think it won't activate as validation if the value is `null`

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/ClaimData.java [hunk 3]
```diff
@@ -173,8 +170,11 @@ public Boolean isClaimantRepresented() {
     }
 
     @JsonIgnore
-    public BigDecimal getFeesPaidInPounds() {
-        return MonetaryConversions.penniesToPounds(new BigDecimal(feeAmountInPennies));
+    public Optional<BigDecimal> getFeesPaidInPounds() {
+        if (feeAmountInPennies == null) {
+            return Optional.empty();
+        }
+        return Optional.ofNullable(MonetaryConversions.penniesToPounds(new BigDecimal(feeAmountInPennies)));
```
Comment (andrewwa-kainos) [inline domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/ClaimData.java]: Can be simplified
```
return Optional.ofNullable(feeAmountInPennies)
    .map(BigDecimal::new)
    .map(MonetaryConversions::penniesToPounds);
```

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java [hunk 4]
```diff
@@ -196,19 +196,19 @@ public CreatePaymentResponse initiatePayment(
 
         Claim createdClaim = caseRepository.initiatePayment(user, claim);
 
+        Optional<Payment> payment = createdClaim.getClaimData().getPayment();
         return CreatePaymentResponse.builder()
-            .nextUrl(createdClaim.getClaimData().getPayment().getNextUrl())
+            .nextUrl(payment.orElseThrow(IllegalStateException::new).getNextUrl())
```
Comment (andrewwa-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java]: May as well throw the exception as soon as we have the optional

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/ClaimData.java (hunk 3 above)
Comment (sabahirfan) [reply domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/ClaimData.java]: done

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java (hunk 4 above)
Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimService.java]: `payment` is being initialised in call back now not coming from front-end.

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/ClaimData.java (hunk 2 above)
Comment (sabahirfan) [reply domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/ClaimData.java]: That is never going to be invoked as `feeAmountInPennies` will never be passed from front-end after IOC. will always be initiated in callback now. Anyways added that back for current scenario.

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/ClaimData.java (hunk 3 above)
Comment (sabahirfan) [reply domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/ClaimData.java]: done

**Pull request merged**